service/service.go

```diff
@@ -0,0 +1,53 @@
+// Package implements a function, that generates a random sequence of parentheses with length = n, and handler,
+// that handle request to /generate?n={length}
+package service
+
+import (
+	"errors"
+	"math/rand"
+	"net/http"
+	"strconv"
+)
+
+var ErrNegativeLength = errors.New("negative length")
+
+// HandleRequests implements all handlers
+func HandleRequests() {
+	http.HandleFunc("/generate", GenerateHandler)
+}
+
+// GenerateHandler handles request to /generate?n={length}
+func GenerateHandler(w http.ResponseWriter, r *http.Request) {
+	n := r.URL.Query().Get("n")
+	l, err := strconv.Atoi(n)
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+	}
+
+	s, err := ParenthesesGeneration(l)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+	}
+
+	_, err = w.Write([]byte(s))
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+// ParenthesesGeneration generates a random sequence of parentheses of n = length
+func ParenthesesGeneration(n int) (string, error) {
+	if n < 0 {
+		return "", ErrNegativeLength
+	}
+
+	p := "[]{}()"
+	g := ""
+
+	for i := 0; i < n; i++ {
+		g += string(p[rand.Intn(len(p))])
```

Comment:
`strings` are immutable. Every time you concat a string, you create a new string. It's bad. 
Concatenating is ok, if you have just a few parts like this:
```go
res := "you have " + count + " new messages"
```
Then you don't suffer much losses in terms of performance, but your code is clear.

In your case where you will concat thousands of times, use string builder